### PR TITLE
Add B3 multiple headers propagation

### PR DESCRIPTION
**Pull request by duxet**
Solves #216.

I had to do some refactoring, as current setup was prepared only for single header propagation.

### CODE REVIEW

File: src/Trace/Propagator/HttpHeaderPropagator.php [hunk 1]
```diff
@@ -75,33 +75,11 @@ public function extract($headers)
      */
     public function inject(SpanContext $context, $container)
     {
-        $header = $this->key();
+        $header = $this->header;
         $value = $this->formatter->serialize($context);
-        if (!headers_sent()) {
```
Comment (duxet) [inline src/Trace/Propagator/HttpHeaderPropagator.php]: i am not sure if i should keep it, it shouldn't cause any side effects IMHO

File: src/Trace/Propagator/B3HeadersPropagator.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    {
+        $traceId = $container[self::X_B3_TRACE_ID] ?? null;
+        $spanId = $container[self::X_B3_SPAN_ID] ?? null;
+        $sampled = $container[self::X_B3_SAMPLED] ?? null;
+        $flags = $container[self::X_B3_FLAGS] ?? null;
+
+        $enabled = null;
+
+        if ($sampled !== null) {
+            $enabled = $sampled === '1';
```
Comment (jcchavezs) [inline src/Trace/Propagator/B3HeadersPropagator.php]: It is possible some propagators pass `true` aswell according to https://github.com/openzipkin/b3-propagation#sampling-state-1

**Comment by nenadstojanovikj** [comment]
@chingor13 @basvanbeek Could you please review this PR?

**Comment by sergiughf** [comment]
+1 looks :ok:  let's get this merged?

**Comment by vgarvardt** [comment]
Seems good.
Can this be merged?

**Comment by nhatthm** [comment]
@chingor13 @basvanbeek Could you please review this PR?

**Pull request closed without merging**